test/parallel/test-stream2-readable-empty-buffer-no-eof.js

```diff
@@ -22,28 +22,27 @@ function test1() {
 
   const buf = Buffer.alloc(5, 'x');
   let reads = 5;
-  const timeout = common.platformTimeout(50);
   r._read = function(n) {
     switch (reads--) {
       case 0:
         return r.push(null); // EOF
       case 1:
         return r.push(buf);
       case 2:
-        setTimeout(r.read.bind(r, 0), timeout);
+        setImmediate(r.read.bind(r, 0));
         return r.push(Buffer.alloc(0)); // Not-EOF!
       case 3:
-        setTimeout(r.read.bind(r, 0), timeout);
+        setImmediate(r.read.bind(r, 0));
         return process.nextTick(function() {
           return r.push(Buffer.alloc(0));
         });
       case 4:
-        setTimeout(r.read.bind(r, 0), timeout);
-        return setTimeout(function() {
+        setImmediate(setImmediate, r.read.bind(r, 0));
```

Comment:
Oh, I misread your comment. Urp. Yeah, I wonder if the thing to do is just swap them like this:

```js
        const rv = setImmediate(function() {
          return r.push(Buffer.alloc(0));
        });
        setImmediate(setImmediate, r.read.bind(r, 0));
        return rv;
```

Easier to read/understand and same effect?